EMBRACE_Client/ServerCommunicationController.m

```diff
@@ -0,0 +1,1265 @@
+    //Setup POST method with proper encoding
+    NSString *post = [NSString stringWithFormat:@"movenum=%ld&objectid=%@&posx=%f&posy=%f", (long)movenum.integerValue, objid, posx, posy];
+    NSData *postData = [post dataUsingEncoding:NSASCIIStringEncoding allowLossyConversion:YES];
+    NSString *postLength = [NSString stringWithFormat:@"%lu", (unsigned long)[postData length]];
+    
+    //Add the necessary URL, Headers and POST Body toNSMutableURLRequest
+    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] init];
+    [request setURL:[NSURL URLWithString:@"http://129.219.28.98/embrace-log-objects.php"]];
+    [request setHTTPMethod:@"POST"];
+    [request setValue:postLength forHTTPHeaderField:@"Content-Length"];
+    [request setValue:@"application/x-www-form-urlencoded;charset=UTF-8" forHTTPHeaderField:@"Content-Type"];
+    [request setHTTPBody:postData];
+    
+    //To send and recieve a response
+    NSURLResponse *response;
+    NSData *POSTReply = [NSURLConnection sendSynchronousRequest:request returningResponse:&response error:nil];
+    NSString *theReply = [[NSString alloc] initWithBytes:[POSTReply bytes] length:[POSTReply length] encoding: NSASCIIStringEncoding];
+    NSLog(@"Reply: %@", theReply);
+    
+   }
+
+//local XML logging starts here
+
+//stores general context of participant, condition and expermenter as global elements
+-(void) logContext : (Student *) userdetails
+{
+    if(userdetails != nil) {
+        [self init]; //start a new log file
+        
+        //formats username string into "firstname lastname"
+        NSString *FileNameValue = [NSString stringWithFormat:@"%@ %@ %@",[userdetails schoolName],[userdetails firstName],[userdetails lastName]];
+        
+        //sets global variables to be used by returnContext function
+        studySchoolString = [userdetails schoolName];
+        studyExperimenterString = [userdetails experimenterName];
+        studyConditionString = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"CFBundleDisplayName"]; //comes from app name
+        studyParticipantString = [userdetails firstName];
+        studyDayString = [userdetails lastName];
+        studyFileName = FileNameValue;
+    }
+
+}
+
+
+//Logging Computer Actions
+//logging object manipulation
+
+
+/*
+ Action Type: Automatic Computer Move Object
+ UserActionIDTag: currentIDTag
+ Selection: ObjID of image being moved
+ Action: Move Object(s)
+ Input: What objects, start location, end location
+ Context: story, chapter, page, sentence, step, username, condition, experimenter
+ */
+-(void) logComputerMoveObject : (NSString *) movingObjectID : (NSString *) collisionObjectorLocationID : (float) startPosX : (float) startPosY : (float) endPosX : (float) endPosY : (NSString *) computerAction : (NSString *) storyValue : (NSString *) chapterValue : (NSString *) pageValue : (NSString *) sentenceValue : (NSString *) stepValue
+{
+    
+    //logging structure for computer actions
```

Comment:
A lot of the code in these logging functions gets repeated. The similar code should be moved into a separate function.